includes/node/multi_queue_node.h

```diff
@@ -13,20 +15,20 @@ class CRMultiQueueNodeBase : public CRNode {
    public:
     explicit CRMultiQueueNodeBase(size_t queue_capacity) : queue_capacity_(queue_capacity) {}
 
-    CRMessageQueue* MessageQueueMapper(const CRMessageBasePtr& message) override;
+    CRMessageQueue* MessageQueueMapper(const channel_id_t channel) override;
 
    protected:
     using queue_callback_t = std::function<void(const CRMessageBasePtr&)>;
+    using queue_map_t = std::unordered_map<channel_id_t, std::unique_ptr<CRMessageQueue>, boost::hash<channel_id_t>>;
```

Comment:
Why do we need the hash from boost for this?

Reply:
Because there are no default hash for `std::pair`